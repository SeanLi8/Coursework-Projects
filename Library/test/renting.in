commands
add collection books.csv
add member Sean Li
add member Neas Il
member 100000
member 100001
rent 100000 999999
rent 999999 111111
member books 100000
member books 100001
rent 100000 111111
member books 100000
member books 100001
rent 100000 111112
rent 100001 111111
member books 100000
member books 100001
relinquish 100000 111111
rent 100001 111111
member books 100000
member books 100001
exit

