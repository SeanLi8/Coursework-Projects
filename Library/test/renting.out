user: EXIT ends the library process
COMMANDS outputs this help string

LIST ALL [LONG] outputs either the short or long string for all books
LIST AVAILABLE [LONG] outputs either the short of long string for all available books
NUMBER COPIES outputs the number of copies of each book
LIST GENRES outputs the name of every genre in the system
LIST AUTHORS outputs the name of every author in the system

GENRE <genre> outputs the short string of every book with the specified genre
AUTHOR <author> outputs the short string of every book by the specified author

BOOK <serialNumber> [LONG] outputs either the short or long string for the specified book
BOOK HISTORY <serialNumber> outputs the rental history of the specified book

MEMBER <memberNumber> outputs the information of the specified member
MEMBER BOOKS <memberNumber> outputs the books currently rented by the specified member
MEMBER HISTORY <memberNumber> outputs the rental history of the specified member

RENT <memberNumber> <serialNumber> loans out the specified book to the given member
RELINQUISH <memberNumber> <serialNumber> returns the specified book from the member
RELINQUISH ALL <memberNumber> returns all books rented by the specified member

ADD MEMBER <name> adds a member to the system
ADD BOOK <filename> <serialNumber> adds a book to the system

ADD COLLECTION <filename> adds a collection of books to the system
SAVE COLLECTION <filename> saves the system to a csv file

COMMON <memberNumber1> <memberNumber2> ... outputs the common books in members' history

user: 26 books successfully added.

user: Success.

user: Success.

user: 100000: Sean Li

user: 100001: Neas Il

user: No such book in system.

user: No such member in system.

user: Member not currently renting.

user: Member not currently renting.

user: Success.

user: To Kill a Mockingbird (Harper Lee)

user: Member not currently renting.

user: Success.

user: Book is currently unavailable.

user: To Kill a Mockingbird (Harper Lee)
The Hobbit (J.R.R. Tolkien)

user: Member not currently renting.

user: Success.

user: Success.

user: The Hobbit (J.R.R. Tolkien)

user: To Kill a Mockingbird (Harper Lee)

user: Ending Library process.
